docs/dev/style-guide.rst

```diff
@@ -67,3 +84,126 @@ but even more importantly,
 it helps and includes readers by giving them quick and easy access to terms that they may be unfamiliar with.
 
 Use an external link or Intersphinx reference when a term is clearly defined elsewhere.
+
+Cross-references
+----------------
+
+Cross-references are great to have as :ref:`inline links <style-guide:Cross-references>`.
+Because of sphinx-hoverxref_,
+inline links also have a nice tooltip displayed.
+
+We like to cross-reference other articles with a definition list inside a ``seealso::`` admonition box.
+It looks like this:
+
+.. code-block:: rst
+
+   .. seealso::
+
+      :doc:`/other/documentation/article`
+        You can learn more about <concept> in this (how-to/description/section/article)
+
+.. _sphinx-hoverxref: https://sphinx-hoverxref.readthedocs.io/
+
+Headlines
+---------
+
+Sphinx is very relaxed about how headlines are applied and will digest different notations.
+We try to stick to the following:
+
+.. code-block:: rst
+
+   Header 1
+   ========
+
+   Header 2
+   --------
+
+   Header 3
+   ~~~~~~~~
+
+   Header 4
+   ^^^^^^^^
+
+In the above, ``Header 1`` is the title of the article.
+
+Diátaxis Framework
+------------------
+
+We apply the methodology and concepts of the Diátaxis Framework.
+This means that *both content and navigation path* for all sections should fit a single category of the 4 Diátaxis categories:
+
+* Tutorial
+* Explanation
+* How-to
+* Reference
+
+.. seealso::
+
+   `https://diataxis.fr/ <https://diataxis.fr/>`__
+     The official website of Diátaxis is the main resource.
+     It's best to check this out before guessing what the 4 categories mean.
+
+.. warning:: **Avoid minimal changes**
+
+   If your change has a high coherence with another proposed or planned change,
+   propose the changes in the same PR.
+
+   By multi-tasking on several articles about the same topic,
+   such as an explanation *and* a how-to,
+   you can easily design your content to end up in the right place *Diátaxis-wise*.
+   This is great for the author and the reviewers
+   and it saves coordination work.
+
+   Minimal or isolated changes generally raise more questions and concerns
+   than changes that seek to address a larger perspective.
```

Comment:
I think in general it's the opposite with code, but for this refactor, it's more true because we want to fix all the issues around a topic at once, instead of leaving them worse than we found them. So not 100% sure this will prove true after the initial refactor, but we'll see.

Reply:
Agreed - I hope that the "Warning" format of this will automatically make it seem opinionated enough that we will want to re-evaluate it.